### DISCUSSION TITLE
Azure function to read pmtiles

### DISCUSSION THREAD

**Discussion by kbatchu** [protomaps/PMTiles#293] [Q&A] [answered]
I was able to successfully setup the Cloudflare integration by following the instructions provided in protomaps docs. I want to replicate this set up in Azure. Do I need to create an Azure function similar to the Cloudflare Worker (provided in the docs) to read the pmtiles files? If yes, is there any sample out there that I can refer to?

Or can I just use ol-pmtiles (OpenLayers) JS library in my web app and load the Azure pmtiles blob files directly?
Thanks,

**Comment by bdon** [comment, marked as answer]
There's an existing PR with a proof of concept here: https://github.com/protomaps/PMTiles/pull/197 but I haven't had the chance to dive deeper on this. It would be great to have first-class support for Azure just like with AWS and Cloudflare. please report back on if you make progress here! 

Yes, you can use ol-pmtiles and load from azure blobs directly via HTTP using the current code; there are some latency + compatibility benefits of using a serverless function though.

**Reply by kbatchu** [reply]
Thanks for the link to the PR. I'll try...but I am not Brandon Liu :-)

**Comment by kbatchu** [comment]
Hi Brandon,
I read the recently updated documentation for deploying pmtiles on Azure with the Azure Container App using the go-pmtiles Docker image. It further says that the tiles can be served through this URL: https://example.name.region.azurecontainerapps.io/tileset/0/0/0.mvt. The part where I am confused is that there is no link in the documentation to the Azure serverless function that receives the request and converts it to a range request similar to the Cloudflare worker javascript function. Is this Azure function part of the go-pmtiles Docker image?
I apologize if this is a lame question. I am still learning the basics of Azure and am not too familiar with docker.
Thanks as always for your brilliant work,
Kiran

**Reply by bdon** [reply]
We are not using serverless functions at all anymore, and using container apps instead which has scale-to-zero. the code is in the docker image, correct, it's just the standard go-pmtiles image.

**Reply by kbatchu** [reply]
Thanks for the clarification. The documentation refers to the use of the go-pmtiles docker image for Google Cloud and Azure only. So I assume the docker image cannot be used for Cloudflare and AWS setup. Is this correct?
Thanks,

**Reply by bdon** [reply]
correct, in theory you could try to run the docker image in aws lambda or cloudflare workers. if you try it out report back how it is!

**Reply by kbatchu** [reply]
Well, for now, we are just trying to make this work in Azure (within a Federal agency)...so this might be a game changer and hopefully, at least I will be able to report back on this.
Thanks so much!

**Reply by kbatchu** [reply]
Hi @bdon 
Update: So I tried setting up the Azure Container App using the go-pmtiles docker image. I followed the instructions listed here: https://docs.protomaps.com/deploy/azure. For command override, I used this: **/go-pmtiles, serve, ., --bucket=azblob://pmtiles?storage_account=grspappdevsa, --public-url=https://bv-test-grsp-container-app.grspcloudnativedev.cdc.gov**.
storage account = grspappdevsa
storage container = pmtiles
I made sure that the root url was working: **https://bv-test-grsp-container-app.grspcloudnativedev.cdc.gov** displays the 'Hello World' page.
I tried accessing a pmtiles file in the blob storage using this url: **https://bv-test-grsp-container-app.grspcloudnativedev.cdc.gov/pmtiles/eji2019_tracts_z2.pmtiles**, but I got a 404 error. 
I talked to a microsoft tech support and I was told that the 'Service connector' way of connecting to blob storage is not recommended since it is in 'preview' and unstable. He recommended using 'storage mounts' instead of 'Service connector.'
Do you have any recommendations or any other ideas on how to make this work in Azure?
Thanks
Kiran
